tasks/check-data-keys/check-data-keys.yaml

```diff
@@ -24,52 +29,46 @@ spec:
       description: The workspace where the data JSON file resides
   steps:
     - name: check-data-keys
-      image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+      image: quay.io/konflux-ci/release-service-utils@sha256:b8a4a4ff239d7789029a7bbd8c89d41453107d4391c1aaf7ea158f343d5624f6
       args: ["$(params.systems[*])"]
       script: |
         #!/usr/bin/env bash
         set -ex
 
-        KEYS_JSON='{
-            "releaseNotes": [
-                "product_id",
-                "product_name",
-                "product_version",
-                "product_stream",
-                "cpe",
-                "type",
-                "content.images",
-                "synopsis",
-                "topic",
-                "description",
-                "solution",
-                "references"
-            ],
-            "cdn": [
-                "env"
-            ]
-        }'
-
         if [ ! -f "$(workspaces.data.path)/$(params.dataPath)" ] ; then
             echo "No data JSON was provided."
             exit 1
         fi
 
-        RC=0
+        if ! curl -s --fail-with-body "$(params.schema)" -o /tmp/output ; then
```

Comment:
It'd be really nice in the mocks.sh of this task to mock curl with like `cat ../../../../schema.json` (not the real path) so that changes to the schema that are part of this PR are included in the pr unit tests

Reply:
I was thinking about this: Would it make more sense for the schema to have its own tests? As the schema can be updated, but the check-data-leys will not be changed, meaning the tests will not run. Also, since the schema is now in another PR, it won't pull the correct schema from local.

I'm finding it hard to pass in the local schema to the Tekton test. As we normally create the JSON with echo inside the task.

Reply:
I don't think it makes sense for a schema to have its own tests. https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/create-advisory/tests/pre-apply-task-hook.sh#L13 tells me that the mocks execute in a directory with the task. So, if you do some `ls` calls and `pwd`, I would bet you can find the path to the schema and then mock the curl to return it

Reply:
> tells me that the mocks execute in a directory with the task

Depends what you mean by mock execution. When `pre-apply-task-hook.sh` is run, you have access to files. So you inject a mock to the task script. But when the task runs (and a mock is hit), you don't have access to any files in the repo - you're in a container.

But I'm not really sure why a task test would need the real schema. A task check should verify that the task does what it should do - in this case it runs schema validation using the schema you supply. So to verify that, using some simple schema should be enough.

Reply:
It is nice to use the real schema because if a PR updates the schema in a way that breaks the test, we would catch it (assuming this task was also updated in the same PR). 
> So you inject a mock to the task script. But when the task runs (and a mock is hit), you don't have access to any files in the repo - you're in a container.

Yeah, so the mock can be like
```
curl() {
    cat ../../../../schema.json
}

Reply:
> Yeah, so the mock can be like

How will the file get to that location in that container that executes the task script in the test?

Reply:
Yeah I talked about this in dms with Sean. If the loadstr thing doesn't end up working, I am fine with him scrapping my idea

Reply:
Ok, I have added one that curls the development. Let me know if we are happy with or we can just remove it.